backend/wallet/services/offchain/payment_command.py

```diff
@@ -1,159 +1,84 @@
-    destination_subaddress: str,
-    amount: int,
-    currency: DiemCurrency,
-) -> offchain.PaymentCommand:
-    sender_onchain_address = context.get().config.vasp_address
-    sender_sub_address = account.generate_new_subaddress(account_id=sender_id)
+def add_payment_command(
+    account_id,
+    reference_id,
+    vasp_address,
+    merchant_name,
+    action,
+    currency,
+    amount,
+    expiration: int,
+) -> None:
+    my_address = generate_my_address(account_id)
 
-    command = offchain.PaymentCommand.init(
-        identifier.encode_account(sender_onchain_address, sender_sub_address, _hrp()),
-        _user_kyc_data(sender_id),
-        identifier.encode_account(destination_address, destination_subaddress, _hrp()),
-        amount,
-        currency.value,
-    )
+    sig_msg = txnmetadata.travel_rule(
+        reference_id, identifier.decode_account_address(my_address, hrp()), amount
+    )[1]
 
-    save_payment_command(
-        payment_command_to_model(command, TransactionStatus.OFF_CHAIN_OUTBOUND)
+    payment_command = models.PaymentCommand(
+        my_actor_address=my_address,
+        inbound=False,
+        cid=str(uuid.uuid4()),
+        reference_id=reference_id,
+        sender_address=my_address,
+        sender_status=Status.none,
+        sender_kyc_data=offchain.to_json(user_kyc_data(account_id)),
+        receiver_address=vasp_address,
+        receiver_status=Status.none,
+        amount=amount,
+        currency=currency,
+        action=action,
+        created_at=datetime.now(),
+        status=TransactionStatus.PENDING,
+        account_id=account_id,
+        merchant_name=merchant_name,
+        expiration=datetime.fromtimestamp(expiration),
+        recipient_signature=compliance_private_key().sign(sig_msg).hex(),
     )
+    save_payment_command(payment_command)
 
-    return command
-
-
-def process_inbound_command(
-    request_sender_address: str, request_body_bytes: bytes
-) -> (int, bytes):
-    command = None
-    try:
-        command = _offchain_client().process_inbound_request(
-            request_sender_address, request_body_bytes
-        )
-        logger.info(f"process inbound command: {offchain.to_json(command)}")
-
-        if command.command_type() == CommandType.PaymentCommand:
-            payment_command = typing.cast(offchain.PaymentCommand, command)
-            _lock_and_save_inbound_command(payment_command)
-        elif command.command_type() == CommandType.FundPullPreApprovalCommand:
-            preapproval_command = typing.cast(
-                offchain.FundsPullPreApprovalCommand, command
-            )
-            handle_fund_pull_pre_approval_command(preapproval_command)
-
-        return _jws(command.id())
-    except offchain.Error as e:
-        logger.exception(e)
-        return _jws(command.cid if command else None, e.obj)
 
+def update_payment_command_status(reference_id, status):
+    storage.update_payment_command_sender_status(reference_id, status)
```

Comment:
For someone reading the code not in the context of the current feature, it might look weird that you are so convinced rather the sender status should be updated. Maybe the service method should have the same name?